web/src/modules/user/wallets/sagas/walletsSaga.ts

```diff
@@ -8,20 +8,24 @@ const walletsOptions: RequestOptions = {
     apiVersion: 'peatio',
 };
 
-const currenciesOptions: RequestOptions = {
-    apiVersion: 'peatio',
-};
+export const getCurrencies = state => state.public.currencies;
+export const getUser = state => state.user.profile.userData.user;;
 
 export function* walletsSaga(action: WalletsFetch) {
     try {
         const accounts = yield call(API.get(walletsOptions), '/account/balances');
-        const currencies = yield call(API.get(currenciesOptions), '/public/currencies');
+        const currenciesList =  yield select(getCurrencies);
+        const user = yield select(getUser);
 
-        yield put(currenciesData(currencies));
+        const currencies: Currency[] = currenciesList.list;
 
         const accountsByCurrencies = currencies.map((currency: Currency) => {
             let walletInfo = accounts.find((wallet: Wallet) => wallet.currency === currency.id);
 
+            if (currency.status === 'hidden' && user.role !== 'admin' && user.role !== 'superadmin') {
```

Comment:
I see you're using this check several times
how about adding a selector isCurrencyAllowed which will check that values directly from store